iceoryx_hoofs/vocabulary/include/iox/detail/string.inl

```diff
@@ -238,6 +238,35 @@ inline bool string<Capacity>::unsafe_assign(const char* const str) noexcept
     return true;
 }
 
+template <uint64_t Capacity>
+inline void string<Capacity>::unsafe_auto_raw_access(
+    const iox::function_ref<void(char*, const iox::BufferInfo info)>& func) noexcept
+{
+    iox::BufferInfo info{m_rawstringSize, Capacity};
+    func(m_rawstring, info);
+
+    const uint64_t strSize{strnlen(m_rawstring, Capacity + 1U)};
+    IOX_EXPECTS_WITH_MSG(Capacity >= strSize,
+                         "unsafe_auto_raw_access failed. Data wrote outside the maximun string capacity of "
+                             << Capacity);
+    m_rawstringSize = strSize;
+}
+
+template <uint64_t Capacity>
+inline void
+string<Capacity>::unsafe_raw_access(const iox::function_ref<uint64_t(char*, const iox::BufferInfo info)>& func) noexcept
+{
+    iox::BufferInfo info{m_rawstringSize, Capacity};
```

Comment:
Please use `Capacity + 1`. This is a low level API and the user might need to write a zero termination at the end of the buffer which is larger than the `Capacity` by one element.